### ISSUE TITLE
Agent name split for 2K names

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#3700]
@dustymc I have just over 2,000 agent names that I want to run through the Agent name splitter, but it times out. Rather than split it up and run the tool a bunch of times, can you run them for me? File attached.
[ASUMZ Agents.zip](https://github.com/ArctosDB/arctos/files/6751172/ASUMZ.Agents.zip)

**Comment by dustymc** [comment]
You can just reload a few times, although that's about as far from obvious as it's possible to be given the behavior...

[splitAgentNames(5).csv.zip](https://github.com/ArctosDB/arctos/files/6751329/splitAgentNames.5.csv.zip)

**Comment by Jegelewicz** [comment]
EVERY name has this:

> FATAL ERROR: Person agents must have at least one of first, middle, last name.

Why? Even good, 3 full name names! Like Jeffery Hunter Sebern

**Comment by dustymc** [comment]
>Why?

That's how that form works - ignore it if it's not useful. I occasionally clean up at that point so I've included the check even though it can't run correctly there, but that's a fairly small and fairly clean dataset so getting those at the next step might make sense.

**Comment by dustymc** [comment]
Oh - and it's always going to alert on "hunter" - it's a real people-name, but I've also probably deleted a thousand variations of "person who hunts" over the years and complaints are way easier to deal with than duplicates.

**Comment by Jegelewicz** [comment]
> You can just reload a few times, although that's about as far from obvious as it's possible to be given the behavior...

Can we turn this into some kind of "I'll notify you when results are ready" thing?

> ignore it if it's not useful.

I'm OK with the hunter thing, but telling me there is a FATAL ERROR, when there isn't one seems a bit over the top. For you and me that may be OK, for the person whose very first interaction with getting data into Arctos (and this is the first step) it will be overwhelming and a bad sign. Also, the addition of a "formatted name" which cannot be loaded in the bulkloader, means I have to delete stuff that is really just a duplicate of "preferred name".

I want to help new collections get their data in and starting off with a big pile of cryptic and scary messages won't help that. A good first step would be to put the "suggestions" message types in separate columns. Right now, they are all mashed together which makes it difficult to sort them and, for instance, find all of the exact matches. Something like this:

PREFERRED_NAME | OTHER_NAME_1 | OTHER_NAME_TYPE_1 | OTHER_NAME_2 | OTHER_NAME_TYPE_2 | OTHER_NAME_3 | OTHER_NAME_TYPE_3 | Missing_name | Exact_match | Close_match | Not_person | Suggest_unknown
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
B. Pannock | Pannock | last name |   | middle name | B. | first name | middle |   |   |   | Do not unnecessarily abbreviate   (preferred_name contains period)
H. H. Miller | Miller | last name | H. | middle name | H. | first name |   | <a   href="/agents.cfm?agent_id=10008365"   target="_blank">H. H. Miller</a> (exact preferred name   match,nocase first and last name match,nodots-nospaces match on agent   name,nodots-nospaces-nocase match on first middle last) | <a   href="/agents.cfm?agent_id=21284469"   target="_blank">H. A. Miller</a> (nocase first and last   name match); <a href="/agents.cfm?agent_id=21272811"   target="_blank">H. C. Miller</a> (nocase first and last   name match) |   | Do not unnecessarily abbreviate   (preferred_name contains period)
R J M | M | last name | J | middle name | R | first name |   |   | <a   href="/agents.cfm?agent_id=21273611"   target="_blank">R. J. M.</a> (nodots-nospaces match on   agent name); <a href="/agents.cfm?agent_id=21291352"   target="_blank">Ricardo J. Gordillo Martinez</a>   (nodots-nospaces match on agent name) |   | Check case of preferred name
Freddy Hunter | Hunter | last name |   | middle name | Freddy | first name | middle |   |   | Check name for hunter |  
Lane |   | last name |   | middle name |   | first name | first, middle, last |   | <a   href="/agents.cfm?agent_id=1810" target="_blank">A.   Lane</a> (nodots-nospaces match on agent name);<a   href="/agents.cfm?agent_id=21290052"   target="_blank">Allen Lane</a> (nodots-nospaces match on   agent name) |   | Do not create unnecessary   variations of `unknown.` (preferred name is one word)

Any chance we could do that?

**Comment by dustymc** [comment]
> "I'll notify you when results are ready" thing?

Yes, but I haven't quite figured out how yet. (I think we may be overloading "component loaders," but they contain the general idea.)

>for you 

That's almost always my first audience, and it's fairly often the last. I think that evolutionary design has worked exceptionally well both in making tools that work for someone other than me and in avoiding wasting resources developing tools that nobody but me will ever use, but maybe we need to be somehow better at categorizing how refined tools are (and encouraging "I ain't you plz make it better" issues) or something.

>that is really just a duplicate of "preferred name".

Only for some use cases. If you're seeing it, I've dealt with data in which is was useful.

>and this is the first step

Off a cliff....

Agents are (sometimes) hellishly complicated, and there are some pretty sophisticated tools that help unwind them in Arctos. I'd rather not dump that on some new collection before they really internalize things like "normalization," even if the tools were more refined. If there's a technically-competent mentor available maybe it does make sense to tackle this up front, but in most cases I think we should probably change the guidance to loading agents (assuming they're coming from something like a column in a spreadsheet - none of this is likely to make sense if there's already some structure) as 'verbatim collector' attributes (there are reasons to do this no matter what else happens) and sorting them out to data objects later - I think all the tools to do so exist, and could be fairly easily streamlined. I'm not suggesting we don't improve tools, just that maybe we need to guide people into them down a slightly different path as we're improving them.

>separate columns

That's a major rebuild of the behind-the-scenes tool, but it seems useful and all the bits speak a language in which it's possible now (they didn't when this was developed) - seems like a good idea to me.

>Any chance 

Yea, I think so. We need to work out the big picture (I'll comment in your other agent issue) and then prioritize.

**Comment by Jegelewicz** [comment]
>     that is really just a duplicate of "preferred name".
> 
> Only for some use cases. If you're seeing it, I've dealt with data in which is was useful.

What I was trying to say is, I can't load "formatted name", so this is taking up space in something I could potentially use to kick off my Agent bulkload.

> Off a cliff....

Realistically this is better done up front. If you know your people names are in Arctos and they match your people names in your bulkload, that is one less bunch of bulkloader errors to deal with. This process works really well:

Agents - make sure they are in Arctos and match your data
Accessions - ditto
Taxon Names - ditto
Taxon Classifications - do all your identification names have associated classifications? Does it matter to you?
Higher Geography - is all of your higher geography in Arctos? Is it formatted correctly in your data?
Locality - Look for near matches in your "places" and harmonize as possible

Bulkload records

Part/Preservation - do your part names and preservations match the code tables?

Bulkload parts

**Comment by dustymc** [comment]
> taking up space

Given some inputs, "preferred names" cannot be accepted by Arctos. This can. It's important, if not for this particular use case.

>better done up front.

That depends on the particulars. This looks like a tiny clean dataset, they're not all like that, cleanup can be incredibly frustrating, we can and should offer alternatives.

**Comment by Jegelewicz** [comment]
> Given some inputs, "preferred names" cannot be accepted by Arctos. This can. It's important, if not for this particular use case.

That is confusing and there is no such thing as "formatted name" in the name type code table...

![image](https://user-images.githubusercontent.com/5725767/124313201-c0d30c00-db2d-11eb-914f-c56b3b89142d.png)

What I am getting at is that it would be nice if the output could be as close to the bulkload tool template as possible....

**Comment by Jegelewicz** [comment]
> This looks like a tiny clean dataset,

HAHAHAHA! not from my perspective! EVERY single one has a FATAL ERROR....

**Comment by campmlc** [comment]
Agree with Teresa. This process needs to be done first, is hair pulling for
mentors, and not a good experience for new collections. There should be no
default FATAL ERROR for any reason. And let's make this more user friendly
please.
